"Verify user is authenticated" Credentials Issue

<!-- Issuehunt Badges -->
[<img alt="Issuehunt badges" src="https://img.shields.io/badge/IssueHunt-%2450%20Funded-%2300A156.svg" />](https://issuehunt.io/r/sindresorhus/np/issues/432)
<!-- /Issuehunt Badges -->


<!--- Provide a short summary of the issue in the title above -->

Hitting an issue with authentication when trying to publish a private scoped package to `npm`.

Running `yarn run release` (where `"release": "np"` in `package.json`):

```
> yarn run release
yarn run v1.16.0
$ np

Publish a new version of @segment/package-name (current: 1.0.4)

Commits:
- Add np for releasing to npm (#3)  66f0afc

Commit Range:
93f42078aea5ad8114c57c618db33f06c89e4147...master

? Select semver increment or specify new version patch 	1.0.5
? This scoped repo @segment/package-name hasn't been published. Do you want to publish it publicly? No

  ❯ Prerequisite check
    ✔ Ping npm registry
    ✔ Check npm version
    ✔ Check yarn version
    ✖ Verify user is authenticated
      → npm ERR!     /Users/colinking/.npm/_logs/2019-06-18T20_37_11_472Z-debug.log
      Check git version
      Check git remote
      Validate version
      Check for pre-release version
      Check git tag existence
    Git
    Cleanup
    Installing dependencies using Yarn
    Running tests using Yarn
    Bumping version using Yarn
    Publishing package using Yarn
    Enabling two-factor authentication
    Pushing tags
    Creating release draft on GitHub

✖ Command failed: npm access ls-collaborators @segment/package-name
npm ERR! code E403
npm ERR! 403 Forbidden - GET https://registry.yarnpkg.com/-/package/%40segment%2Fpackage-name/collaborators?format=cli - Forbidden

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/colinking/.npm/_logs/2019-06-18T20_37_11_472Z-debug.log



Publish failed. Rolling back to the previous state…
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

What's odd is that I am logged in:

```
> yarn login
yarn login v1.16.0
info npm username: colinking
info npm email: <email>
✨  Done in 0.06s.
```

And I'm even able to fetch that collaborators list that is 403-ing:

```
> npm access ls-collaborators @segment/node-ctlstore
{
	// ...
}
```

And both of the following publish fine:

```
> ./node_modules/.bin/np
```

```
> npm run release
```

Possibly this is an issue with `np` getting auth rights correctly via `yarn`?

## Expected behavior

Expected to not get a 403 issue from `np`. It should have published fine, just like it does when run with `npm` instead of `yarn`.

## Environment

np - 5.0.3
Node.js - v10.15.3
npm - 6.9.0
Git - 2.20.1
OS - macOS 10.14.5


<!-- Issuehunt content -->

---

<details>
<summary>
<b>IssueHunt Summary</b>
</summary>


### Backers (Total: $50.00)

- [<img src='https://avatars1.githubusercontent.com/u/9523719?v=4' alt='kylemh' width=24 height=24> kylemh](https://issuehunt.io/u/kylemh) ($50.00)

### Submitted pull Requests
- [#498 Added yarn issue with failing user authentication to the docs](https://issuehunt.io/r/sindresorhus/np/pull/498)
---

#### [Become a backer now!](https://issuehunt.io/r/sindresorhus/np/issues/432)
#### [Or submit a pull request to get the deposits!](https://issuehunt.io/r/sindresorhus/np/issues/432)
### Tips

- Checkout the [Issuehunt explorer](https://issuehunt.io/r/sindresorhus/np/) to discover more funded issues.
- Need some help from other developers? [Add your repositories](https://issuehunt.io/r/new) on IssueHunt to raise funds.
---
IssueHunt has been backed by the following sponsors. [Become a sponsor](https://issuehunt.io/membership/members)
</details>
<!-- /Issuehunt content-->

> And I'm even able to fetch that collaborators list that is 403-ing:

This is the error I get when trying to execute the command:

```
❯ npm access ls-collaborators @segment/package-name
npm ERR! code E403
npm ERR! 403 Forbidden - GET http://registry.npmjs.org/-/package/%40segment%2Fpackage-name/collaborators?format=cli - Forbidden

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/Itai/.npm/_logs/2019-06-24T06_55_25_767Z-debug.log
```

This also might be related to #395, #317 and a few other similar issues.

--- a/readme.md
+++ b/readme.md
@@ -260,6 +260,29 @@ Host *
 
 If you're running into other issues when using SSH, please consult [GitHub's support article](https://help.github.com/articles/connecting-to-github-with-ssh/).
 
+## FAQ
+
+### Publishing with yarn
+#### Question
+When running `np` through yarn, i get the following error:
+```shell
+❯ Prerequisite check
+    ✔ Ping npm registry
+    ✔ Check npm version
+    ✔ Check yarn version
+    ✖ Verify user is authenticated
+
+npm ERR! code E403
+npm ERR! 403 Forbidden - GET https://registry.yarnpkg.com/-/package/my_awesome_package/collaborators?format=cli - Forbidden
+```
+#### Answer
+Please check if `npm access ls-collaborators my_awesome_package` succeeds.
+If so, yarn has overwritten your registry url. Please add the correct registry url in your `package.json`.
+```json
+"publishConfig":{
+	"registry": "https://registry.npmjs.org"
+}
+```
 
 ## Maintainers
 

--- a/readme.md
+++ b/readme.md
@@ -262,22 +262,23 @@ If you're running into other issues when using SSH, please consult [GitHub's sup
 
 ## FAQ
 
-### Publishing with yarn
-#### Question
-When running `np` through yarn, i get the following error:
+### I get an error when publishing my package through Yarn
+
+If you an error like this…
+
 ```shell
 ❯ Prerequisite check
-    ✔ Ping npm registry
-    ✔ Check npm version
-    ✔ Check yarn version
-    ✖ Verify user is authenticated
+✔ Ping npm registry
+✔ Check npm version
+✔ Check yarn version
+✖ Verify user is authenticated
 
 npm ERR! code E403
-npm ERR! 403 Forbidden - GET https://registry.yarnpkg.com/-/package/my_awesome_package/collaborators?format=cli - Forbidden
+npm ERR! 403 Forbidden - GET https://registry.yarnpkg.com/-/package/my-awesome-package/collaborators?format=cli - Forbidden
 ```
-#### Answer
-Please check if `npm access ls-collaborators my_awesome_package` succeeds.
-If so, yarn has overwritten your registry url. Please add the correct registry url in your `package.json`.
+
+…please check whether the command `npm access ls-collaborators my-awesome-package` succeeds. If it does, Yarn has overwritten your registry URL. To fix this, add the correct registry URL to `package.json`:
+
 ```json
 "publishConfig":{
 	"registry": "https://registry.npmjs.org"

